### PR TITLE
refactor: format expiration date for the ignored details panel [IDE-489]

### PR DESCRIPTION
**Pull request by cat2608**
### Description

This PR improves the handling and formatting of expiration dates to ensure they are displayed correctly in the UI. 

### Checklist

- [ ] Tests added and all succeed
- [ ] Linted
- [ ] README.md updated, if user-facing
- [ ] License file updated, if new 3rd-party dependency is introduced

🚨After having merged, please update the CLI go.mod to pull in latest language server.

### Screenshots / GIFs

|Before|After|
|-|-|
|image|image|

### CODE REVIEW

File: infrastructure/code/code_html.go [hunk 1]
```diff
@@ -246,6 +246,26 @@ func getRepoName(commitURL string) string {
...
+	if expiration == "" {
+		return "No expiration"
+	}
+	parsedDate, err := time.Parse(time.RFC3339, expiration)
+	if err != nil {
+		return expiration // Original string if parsing fails
+	}
+
+	// Calculate the difference in days
+	daysRemaining := int(parsedDate.Sub(time.Now()).Hours() / 24)
```
Comment (teodora-sandu) [inline infrastructure/code/code_html.go:259]: Have you tried doing what the GitHub action suggests? Using `time.Until`

File: infrastructure/code/code_html.go [hunk 2]
```diff
@@ -246,6 +246,26 @@ func getRepoName(commitURL string) string {
...
+
+	if daysRemaining < 0 {
+		return "Expired"
```
Comment (teodora-sandu) [inline infrastructure/code/code_html.go:262]: I would expect that if the ignore has expired then it doesn't get returned but it doesn't hurt to have this

**Review by teodora-sandu — APPROVED** [review]

File: infrastructure/code/code_html.go (hunk 2 above)
Comment (cat2608) [reply infrastructure/code/code_html.go:262]: That's true... 🤦‍♀️ 
Thank you for pointing this out 🙏

File: infrastructure/code/code_html.go (hunk 1 above)
Comment (cat2608) [reply infrastructure/code/code_html.go]: I just saw the linter recommendation 💪 
I fixed it 🙌

**Pull request merged**